pkg/server/server.go

```diff
@@ -291,6 +299,45 @@ func WithMaxAuthorizationModelSizeInBytes(size int) OpenFGAServiceV1Option {
 	}
 }
 
+// WithDispatchThrottlingCheckResolverEnabled sets whether dispatch throttling is enabled.
+// Enable this feature will prioritize processing simple check and list objects requests
+// (requests with low number of dispatches) over complex check and list objects
+// (requests with high number of dispatches).
+func WithDispatchThrottlingCheckResolverEnabled(enabled bool) OpenFGAServiceV1Option {
+	return func(s *Server) {
+		s.dispatchThrottlingCheckResolverEnabled = enabled
+	}
+}
+
+// WithDispatchThrottlingCheckResolverTimerTickerFrequency defines how frequent dispatch throttling will be evaluated.
+// TimeTickerFrequency controls how frequently throttled dispatch requests are evaluated to determine whether
+// it can be processed.
+// This value should not be too small (i.e., in the ns ranges) as i) there are limitation in timer resolution
+// and ii) very small value will result in frequent processing of
+func WithDispatchThrottlingCheckResolverTimerTickerFrequency(frequency time.Duration) OpenFGAServiceV1Option {
+	return func(s *Server) {
+		s.dispatchThrottlingCheckResolverTimerTickerFrequency = frequency
+	}
+}
+
+// WithDispatchThrottlingCheckResolverLevel define the number of dispatches to be considered in
```

Comment:
> WithDispatchThrottlingCheckResolverLevel defines the level or threshold that has to be met for dispatch throttling to apply.

I feel we can drop the rest of this godoc comment, because we don't have a notion of high priority or low priority. We either have requests that are throttled or we dont.

Reply:
updated comment